sda/cmd/api/api.go

```diff
@@ -389,3 +392,53 @@ func listUserFiles(c *gin.Context) {
 	c.Writer.Header().Set("Content-Type", "application/json")
 	c.JSON(200, files)
 }
+
+// addHashedKey handles the addition of a hashed key to the database.
+// It expects a JSON payload containing the key hash and its description.
+// If the JSON payload is invalid, it responds with a 400 Bad Request status.
+// If there is no row update in the database, it responds with a 409 Conflict status
+// If the database insertion fails, it responds with a 500 Internal Server Error status.
+// On success, it responds with a 200 OK status.
+func addHashedKey(c *gin.Context) {
```

Comment:
```suggestion
func addC4ghKey(c *gin.Context) {
```

We don't want to calculate the keyhash manually as it will be error prone.
We should send the entire public key and have the API do the hashing.

Reply:
But this function will receive the public key and do the hash calculation and then insert the hash to the database, right? In this case, I think we can still call it `addHashedKey` or `addKeyHash`.